server/events/project_finder.go

```diff
@@ -145,18 +148,34 @@ func (p *DefaultProjectFinder) DetermineProjectsViaConfig(log *logging.SimpleLog
 	return projects, nil
 }
 
-// filterToTerraform filters non-terraform files from files.
-func (p *DefaultProjectFinder) filterToTerraform(files []string) []string {
+// filterToFileList filters out files not included in the file list
+func (p *DefaultProjectFinder) filterToFileList(log *logging.SimpleLogger, files []string, fileList string) ([]string, error) {
 	var filtered []string
-	fileNameRe, _ := regexp.Compile(`^.*(\.tf|\.tfvars|\.tfvars.json)$`)
+	if fileList == "" {
+		fileList = "**/*.tf,**/*.tfvars,**/*.tfvars.json"
+	}
+
+	patterns := strings.Split(fileList, ",")
+	patternMatcher, err := fileutils.NewPatternMatcher(patterns)
+	if err != nil {
+		return nil, errors.Wrapf(err, "filtering modified files with patterns: %v", patterns)
+	}
 
 	for _, fileName := range files {
-		if !p.shouldIgnore(fileName) && (fileNameRe.MatchString(fileName) || filepath.Base(fileName) == "terragrunt.hcl") {
+		if p.shouldIgnore(fileName) {
+			continue
+		}
+		match, err := patternMatcher.Matches(fileName)
+		if err != nil {
+			log.Debug("filter err for file %q: %s", fileName, err)
+			continue
+		}
+		if match || filepath.Base(fileName) == "terragrunt.hcl" {
```

Comment:
since the syntax changed to dockerignore syntax can we just add `**/terragrunt.hcl` to the fileList? I might be wrong, but wouldn't that look for the exact file as well?

Reply:
You are correct. I'll remove the or clause here and add terragrunt.hcl to the defaults.